stathat.go

```diff
@@ -84,6 +85,12 @@ type Reporter struct {
 // default http transport.
 func NewReporter(bufferSize, poolSize int, transport http.RoundTripper) *Reporter {
 	r := new(Reporter)
+	if transport == nil {
+		transport = &http.Transport{
```

Comment:
Can you update the function comment to mention the increased parameter?

Also worth mentioning in the function comment that if you pass in your own transport, it's a good idea to have its MaxidleConnsPerHost be at least the poolSize, and why.